test/OpenTelemetry.Tests/Trace/SamplersTest.cs

```diff
@@ -252,5 +252,29 @@ public void SamplersDoesNotImpactTraceStateWhenUsingNull(SamplingDecision sampli
                 Assert.Equal(parentTraceState, activity.TraceStateString);
             }
         }
+
+        [Fact]
+        public void SamplerExceptionBubblesUpTest()
+        {
+            // TODO: Discuss: An exception thrown in custom sampler probably
+            // should not blow up like this?
+
+            var activitySourceName = Utils.GetCurrentMethodName();
+            using var activitySource = new ActivitySource(activitySourceName);
+            using var tracerProvider = Sdk.CreateTracerProviderBuilder()
+                .AddSource(activitySourceName)
+                .SetSampler(new ThrowingSampler())
+                .Build();
+
+            Assert.Throws<InvalidOperationException>(() => activitySource.StartActivity("ThrowingSampler"));
+        }
+
+        private sealed class ThrowingSampler : Sampler
+        {
+            public override SamplingResult ShouldSample(in SamplingParameters samplingParameters)
+            {
+                throw new InvalidOperationException("ThrowingSampler");
```

Comment:
I think the direction about who is responsible for catching the exception should come from the spec. Similar to how they have specified it for Processor `OnStart` and `OnEnd` methods [here](https://github.com/open-telemetry/opentelemetry-specification/blob/1e48fc558e7e259cccba097ca61880eae98d708b/specification/trace/sdk.md#onendspan).

If we have a particular preference, we should try to get the spec updated for [ShouldSample ](https://github.com/open-telemetry/opentelemetry-specification/blob/1e48fc558e7e259cccba097ca61880eae98d708b/specification/trace/sdk.md#shouldsample) first.

Reply:
> I think the direction about who is responsible for catching the exception should come from the spec. Similar to how they have specified it for Processor `OnStart` and `OnEnd` methods [here](https://github.com/open-telemetry/opentelemetry-specification/blob/1e48fc558e7e259cccba097ca61880eae98d708b/specification/trace/sdk.md#onendspan)

The spec doesn't seem to talk about who should be responsible if exception was thrown from a Processor's `OnStart`/`OnEnd`?

Reply:
> OnEnd is called after a span is ended (i.e., the end timestamp is already set). This method MUST be called synchronously within the [Span.End() API](https://github.com/open-telemetry/opentelemetry-specification/blob/1e48fc558e7e259cccba097ca61880eae98d708b/specification/trace/api.md#end), therefore **it should not block or throw an exception**.

I interpret this statement as "Processor.OnEnd should not throw an exception" which makes it the user's responsibility to provide a spec-compliant processor to the SDK. If the processor logic can throw an exception, it's the user's responsibility to ensure that it is not thrown (they could simply wrap a try-catch around their processor logic.